provisioner.go

```diff
@@ -322,75 +334,68 @@ func (p *LocalPathProvisioner) createHelperPod(action ActionType, cmdsForPath []
 		// it covers the `/` case
 		return fmt.Errorf("invalid path %v for %v: cannot find parent dir or volume dir", action, path)
 	}
-
 	hostPathType := v1.HostPathDirectoryOrCreate
-	helperPod := &v1.Pod{
-		ObjectMeta: metav1.ObjectMeta{
-			Name:      string(action) + "-" + name,
-			Namespace: p.namespace,
-		},
-		Spec: v1.PodSpec{
-			RestartPolicy:      v1.RestartPolicyNever,
-			NodeName:           node,
-			ServiceAccountName: p.serviceAccountName,
-			Tolerations: []v1.Toleration{
-				{
-					Operator: v1.TolerationOpExists,
+	lpvVolumes := []v1.Volume{
+		{
+			Name: "data",
+			VolumeSource: v1.VolumeSource{
+				HostPath: &v1.HostPathVolumeSource{
+					Path: parentDir,
+					Type: &hostPathType,
 				},
 			},
-			Containers: []v1.Container{
-				{
-					Name:    "local-path-" + string(action),
-					Image:   p.helperImage,
-					Command: append(cmdsForPath, filepath.Join("/data/", volumeDir)),
-					VolumeMounts: []v1.VolumeMount{
+		},
+		{
+			Name: "script",
+			VolumeSource: v1.VolumeSource{
+				ConfigMap: &v1.ConfigMapVolumeSource{
+					LocalObjectReference: v1.LocalObjectReference{
+						Name: p.configMapName,
+					},
+					Items: []v1.KeyToPath{
 						{
-							Name:      "data",
-							ReadOnly:  false,
-							MountPath: "/data/",
+							Key:  "setup",
+							Path: "setup",
 						},
 						{
-							Name:      "script",
-							ReadOnly:  false,
-							MountPath: "/script",
-						},
-					},
-					ImagePullPolicy: v1.PullIfNotPresent,
-				},
-			},
-			Volumes: []v1.Volume{
-				{
-					Name: "data",
-					VolumeSource: v1.VolumeSource{
-						HostPath: &v1.HostPathVolumeSource{
-							Path: parentDir,
-							Type: &hostPathType,
-						},
-					},
-				},
-				{
-					Name: "script",
-					VolumeSource: v1.VolumeSource{
-						ConfigMap: &v1.ConfigMapVolumeSource{
-							LocalObjectReference: v1.LocalObjectReference{
-								Name: p.configMapName,
-							},
-							Items: []v1.KeyToPath{
-								{
-									Key:  "setup",
-									Path: "setup",
-								},
-								{
-									Key:  "teardown",
-									Path: "teardown",
-								},
-							},
+							Key:  "teardown",
+							Path: "teardown",
 						},
 					},
 				},
 			},
 		},
 	}
+	lpvVolumeMounts := []v1.VolumeMount{
+		{
+			Name:      "data",
+			ReadOnly:  false,
+			MountPath: parentDir,
+		},
```

Comment:
A CLI option to specify the `MountPropagation` would also be great here.

Reply:
@yasker what do you think?

Reply:
Alternatively, following the template approach, the Pod template could specify the volume mount as well and the code here  could therefore only add the hardcoded volume mount when it is not specified.

Reply:
I don't want to block this PR. So if you'd merge it as is I'll create another PR in which I'd also add a corresponding usage example to clarify the feature's purpose.

Reply:
> I don't want to block this PR. So if you'd merge it as is I'll create another PR in which I'd also add a corresponding usage example to clarify the feature's purpose.

i will add this flag soon, and  @yasker what do you think?

Reply:
> Alternatively, following the template approach, the Pod template could specify the volume mount as well which is then merged with/over the default, hard-coded volume mount above.
> 
> _It would help my use case already if you'd simply set `MountPropagation=v1.MountPropagationBidirectional` but I am not sure whether this could cause issues with other helpers and the default helper doesn't need it._

i add nodePathType and nodePathMountPropagation flag to enable this feature, you can check it. and @yasker what do you think?